src/Npgsql/NpgsqlConnection.cs

```diff
@@ -1012,27 +1012,14 @@ internal void OnNotification(NpgsqlNotificationEventArgs e)
     internal bool IsScramPlus => CheckOpenAndRunInTemporaryScope(c => c.IsScramPlus);
 
     /// <summary>
-    /// Selects the local Secure Sockets Layer (SSL) certificate used for authentication.
+    /// When using SSL/TLS, this is a callback that allows customizing SslStream's authentication options.
     /// </summary>
     /// <remarks>
-    /// See <see href="https://msdn.microsoft.com/en-us/library/system.net.security.localcertificateselectioncallback(v=vs.110).aspx"/>
-    /// </remarks>
-    public ProvideClientCertificatesCallback? ProvideClientCertificatesCallback { get; set; }
-
-    /// <summary>
-    /// When using SSL/TLS, this is a callback that allows customizing how the PostgreSQL-provided certificate is verified. This is an
-    /// advanced API, consider using <see cref="SslMode.VerifyFull" /> or <see cref="SslMode.VerifyCA" /> instead.
-    /// </summary>
-    /// <remarks>
-    /// <para>
-    /// Cannot be used in conjunction with <see cref="SslMode.Disable" />, <see cref="SslMode.VerifyCA" /> and
-    /// <see cref="SslMode.VerifyFull" />.
-    /// </para>
     /// <para>
-    /// See <see href="https://msdn.microsoft.com/en-us/library/system.net.security.remotecertificatevalidationcallback(v=vs.110).aspx"/>.
+    /// See <see href="https://learn.microsoft.com/en-us/dotnet/api/system.net.security.sslclientauthenticationoptions?view=net-8.0"/>.
     /// </para>
     /// </remarks>
-    public RemoteCertificateValidationCallback? UserCertificateValidationCallback { get; set; }
```

Comment:
What do you mean, I've already added `SslClientAuthenticationOptionsCallback ` just below. The main reason I removed previous callbacks (instead of deprecating them) is that then we would have to automatically convert them to `SslClientAuthenticationOptionsCallback ` (just like it's done in `NpgsqlDataSourceBuilder`) which is quite a bit of code. Not really opposed to bring them back, just not sure whether it's really worth it given that the change on the user's side should be quite simple.